tests/test_build.py

```diff
@@ -281,6 +281,28 @@ def test_sidebars_level2(sphinx_build_factory, file_regression):
     file_regression.check(sidebar.prettify(), extension=".html")
 
 
+def test_sidebars_show_nav_level0_index(sphinx_build_factory, file_regression):
+    """"""
+    confoverrides = {"html_theme_options.show_nav_level": 0}
+    sphinx_build = sphinx_build_factory("sidebars", confoverrides=confoverrides).build()
+
+    # Both the column alignment and the margin should be changed
+    index_html = sphinx_build.html_tree("section1/index.html")
+    sidebar = index_html.select("nav#bd-docs-nav")[0]
+    file_regression.check(sidebar.prettify(), extension=".html")
```

Comment:
Instead of doing full regression checks for each of these, why not test specifically for the things that you want to be the case. I think that will reduce the extra cruft in the regression tests and make this test more explicit. So:

- Test that the toctree has a top-level `ul` with a class `list-caption`
- That the next `li` has `toctree-l0 has-children`
- Some basic checks that everything inside has the same structure as we'd expect, but don't need to go crazy here.